### DISCUSSION TITLE
Old graph view

### DISCUSSION THREAD

**Discussion by aipatr** [apache/airflow#35407] [Q&A]
Dear all,
I personally find the new graph view worse than the previous one. Moreover, the fact that every time that I press on the Graph button I get a sub view of the main view and not the entire page is quite annoying, specially with big dags.
Do you know if it possible somehow to have the old view in Airflow 2.7.x ? Thank you in advance

**Comment by thaysa** [comment]
Adding to the current conversation regarding version 2.7.1 experience: The visualization of large DAGs appears to be better in the old layout. Viewing large DAGs in a subview can be challenging. Do you plan to include the graph view in a dedicated window?  Also, are you considering enabling the old graph view?

**Comment by Taragolis** [comment]
AFAIK there is no plan to return old graph view and there is no official discussion for that exists, so I would recommend to create this discussion into to [Apache Airflow devlist](https://airflow.apache.org/community/) - `dev@airflow.apache.org`
- [Apache: Mailing Lists](https://www.apache.org/foundation/mailinglists)
- [Tips for Apache project email contributors](https://infra.apache.org/contrib-email-tips.html)

**Reply by aipatr** [reply]
thank you very much

**Comment by potiuk** [comment]
> The visualization of large DAGs appears to be better in the old layout. Viewing large DAGs in a subview can be challenging. Do you plan to include the graph view in a dedicated window? Also, are you considering enabling the old graph view?

Define better.

**Reply by potiuk** [reply]
Also define challenging.

**Reply by aipatr** [reply]
it is English https://www.vocabulary.com/ Or should I define also the word layout?
With a big dag it is not readable with the new view one cannot read the operator names or see well. The old one was allowing much bigger dag than the current one. If you have simply to troll me with these questions please avoid to comment.

**Reply by potiuk** [reply]
> With a big dag it is not readable with the new view one cannot read the operator names or see well. The old one was allowing much bigger dag than the current one. If you have simply to troll me with these questions please avoid to comment.

No. I am not trolling. I am just not reading in your mind. Many of our users found the new view way better and more functional. Yet you say 'challenging`.  Which might mean many different things for many people - because every one is different and every one has a different perceptions and customs and things that are "challenging" for some migh not exist for others or simply others might not find it challenging. If you try to exercise empathy and understand that different people see things differently and might have different perception of things you will understand that your word "better" and "challenging" can be interpreted in about 1000 different ways by 1000 different people reading them.

My proposal:

Get several scrrenshots or videos - comparing before and after - showing what you find "better" and what you find challenging - possibly explaining it additinally and putting it in context.

When you do it, there is a chance, that people looking at it will try to find a way how to address your concerns and maybe add some improvements that make the current UI even "better", and far less "challenging". But until you do it, there is absolutely no way for those people to find out what is "better" and what is "challenging.

Also I think that would show appreciation to people who spend their night and weekends and release software that you can use for free without any guarantees. And if you understand that the best way you can give back is to clearly specify what problems you have and explain what could be improved, but instead you demand somethign that has long been abandoned to go back - that is what makes being part of the community of the project valuable.

And little more empathetic and seeing that there are humans on the other side as well. 

And yes, it will cost a bit more from your side to exxplain it better, maybe even some time spent on it. And yes. this is the best way you can thank those people who create airflow - and even count yourself among contributors - not even by contributing some code, but maybe contributing ideas on how Airflow could be improved.

I strongly encourage you to consider being empathetic and helpful for the community.

**Reply by aaronolszewski** [reply]
@potiuk @Taragolis Jumping in here to highlight the pain points we are having with the new Graph View. I have included screenshots of the exact same DAG using `2.5.0` and `2.7.0` so that you have context to go by.

As you can see in the first screenshot below, the `task_id` is fully visible (not truncated) and it's very easy to see what `STATE` the task is in as its represented via the colour of the outer border. 
<img width="1721" alt="Screenshot 2023-11-28 at 9 06 26 PM" src="https://github.com/apache/airflow/assets/15781242/1de16fce-48f1-411c-b8bb-5a03d9634483">

In the second screenshot, the `task_id` is getting truncated making it really difficult to see what task is running (especially when quickly glancing at the Graph View). It's also difficult to see what `STATE` a task is in given that the outer border colour has been removed (it would be great if this was added back in)
![dag 2 7 0](https://github.com/apache/airflow/assets/15781242/13063e11-3c6c-49b9-b6ef-2ff8dda4988f)

I do want to say that the new `Cluster Activity` is a fantastic addition. It would be great to add some filters to those charts (e.g. filter by `owner` for example as that would allow owning teams on a shared cluster to filter only their DAGs)

Thanks in advance

**Reply by potiuk** [reply]
Can you please open an issue for that pou articular issue ? I am not doing UI work myself but If you. open a concrete issue you have with the new graph being "long task names are cut - I am sure it will drag attention of those who work on it to see if they can improve it.

**Reply by potiuk** [reply]
Btw. This what I mean by "define" and "challenging" @aipatr . Stating the problem., explaining it. Opening an issue with screenshots showin what you could do better in the old view and what is worse in the new one. 

Doing all that does not guarantee that your problem will be solved, but it will provide concrete input to those who work on UI and can improve it in the future.

**Comment by aipatr** [comment]
It is funny that other users understood it from the description and you had issues with my request. It is not the first time...

Il 29 Nov 2023 19:44, Jarek Potiuk ***@***.***> ha scritto:

Btw. This what I mean by "define" and "challenging" @aipatr<https://github.com/aipatr> . Stating the problem., explaining it. Opening an issue with screenshots showin what you could do better in the old view and what is worse in the new one.

Doing all that does not guarantee that your problem will be solved, but it will provide concrete input to those who work on UI and can improve it in the future.

**Comment by potiuk** [comment]
> It is funny that other users understood it from the description and you had issues with my request. It is not the first time...

It's funny how you do not understand what I was asking you for when other users did. It's not the first time.

**Comment by dejlek** [comment]
@potiuk , here is a bystander perspective for you...

- The first thing that my colleagues asked when I asked them to try new Airflow (I am trying to migrate from 2.6.3 to 2.10.3) was to *bring back the old graph view* because the new one is terrible. 

My view: 2.6.3 had them both, I see no reason why the old graph view could not stay there where it was.

**Reply by aipatr** [reply]
I completely agree. The new view is not really user friendly

**Reply by potiuk** [reply]
@dejlek

> The first thing that my colleagues asked when I asked them to try new Airflow (I am trying to migrate from 2.6.3 to 2.10.3) was to bring back the old graph view because the new one is terrible.

Yep. This is absolutely normal and expected reaction. People generally don't like changes and naturally oppose against them. Also anecdotal evidences are just that - anecdotal evidences. I personally know many users (also new users who had not been accustomed to the old graph) who simply LOVE the new UI. Also anecdotal. So sucha comments without statistics and panels have very little value other than - exactly `anecdotal` value.

@dajlek @aipatr 

Then I will repeat myself. If you have ideas how to improve it -> feel free to open issues for that describing what is missing and what can be improved. In case you have not noticed - Airflow 3.1 will have VASTLY improved UI on grid and graph with numerous improvements implemented - bringing Gantt Chart back and **many** other improvements. All of those improvements were possible PRECISELY because the old graph was dumped and the people who implemented the new UI fixes and improvements could focus on ONE set of changes rather than duplicating the effforts. 

If you look at the recent discussion on the PR of the month https://lists.apache.org/thread/gpr5jcd69mh50vy61b3j1c7pc9b525b7 - about 3 or for great improvements of the UI that will come in 3.1 are viciously fighting for the "PR of the month" label. And all that is possible, because the people who implement those changes could focus on "future" rather than on "bringing the old back".

My question is to both of you - how much did you contribute to discussions about those changes happening ? 

There were many in the PRs where various improvements have been discussed, previewed and people could comment on those. How many of those comments from your side were "this is not the best - let's improve it this or that way".  Many of the people who contributed those are new contributors who never contributed before. People like you who - when they saw something needs improvement - rolled their sleeves up and did it.

I have not seen any of the people commenting here in those discussions. All I hear is "Bring back the old view" - which basically means "stop all the work you are doing on the new UI and improvements and focus on bringing the exact same experiences tha you dumped in order to be able to get better UI faster". Generally you are asking all those new contributors that are now improving the new UI to spend 2x more effort on testing and implementing new stuff that is needed at the backend to support those new changes.

> I see no reason why the old graph view could not stay there where it was.

This is the reason ^^. It would slows improvements of he new UI. That's a reason that is worth it on its own.]


I look forward to your "expectations" that old graph is back, iturning nto proposals and constructive comments on how the new UI will be improved. There is no road back. We boldly look into the future, and you can have a hand in shaping this future rather than demanding those who actually 'do this' - to slow down.
